### PR TITLE
feat: improve tinder for emitter

### PR DESCRIPTION
**Pull request by gfanton**
Fixes https://github.com/berty/zenhub/issues/16

### CODE REVIEW

File: go/internal/tinder/driver_rdvp.go [hunk 1]
```diff
@@ -189,114 +179,56 @@ func (c *rendezvousDiscovery) syncFindPeers(ctx context.Context, ns string, limi
 	return chPeer, err
 }
 
-// nolint:unused
-func (c *rendezvousDiscovery) asyncFindPeers(ctx context.Context, ns string, limit int) (<-chan peer.AddrInfo, error) {
-	// Get cached peers
-	var cache *rpCache
+func (c *rendezvousDiscovery) Subscribe(ctx context.Context, topic string, opts ...discovery.Option) (<-chan peer.AddrInfo, error) {
+	// Get options
+	// var options discovery.Options
```
Comment (D4ryl00) [inline go/internal/tinder/driver_rdvp.go:184]: clean?

File: go/pkg/bertyprotocol/tinder_swiper.go [hunk 2]
```diff
@@ -178,44 +167,44 @@ func (s *Swiper) WatchTopic(ctx context.Context, topic, seed []byte) <-chan peer
...
-			tinder.WatchdogDiscoverForceOption,
-		)
-		if err != nil {
-			cancel()
-			return fmt.Errorf("unable to find peers: %w", err)
-		}
+func (s *Swiper) watchPeers(ctx context.Context, bstrat discovery.BackoffStrategy, out chan<- peer.AddrInfo, topic string) error {
+	sub := s.tinder.Subscribe(topic)
+	defer sub.Close()
+	// func () {
```
Comment (D4ryl00) [inline go/pkg/bertyprotocol/tinder_swiper.go:173]: Should we clean?

File: go/internal/tinder/service.go [hunk 3]
```diff
@@ -3,152 +3,141 @@ package tinder
...
-			s.logger.Warn("unable to close network notify", zap.Error(err))
-		}
-	})
-
-	return nil
+func (s *Service) Close() error {
+	return s.networkNotify.Close()
 }
+
+func (s *Service) GetProcess() uint32 { return atomic.LoadUint32(&s.process) }
```
Comment (D4ryl00) [inline go/internal/tinder/service.go:141]: I don't find where this public method is used.

File: go/internal/rendezvous/emitterio_sync_test.go [hunk 4]
```diff
@@ -25,50 +27,56 @@ func MakeRendezvousServiceTest(ctx context.Context, host host.Host, path string,
...
 	}
 	return clients
 }
 
 func TestEmitterIOFlow(t *testing.T) {
-	t.Skip()
-
-	const serverAddr = "tcp://127.0.0.1:8080"
-	const adminKey = "Llo99jIV4vmN0BqJJO1Z5Y055DsPHnIS"
+	// const adminKey = "bPcKwpuIP3gjjAKqb9EaAJYJnilISQhJ"
```
Comment (D4ryl00) [inline go/internal/rendezvous/emitterio_sync_test.go:45]: remove the adminKey?

File: tool/emitter-server/docker-compose.yml [hunk 5]
```diff
@@ -6,5 +6,5 @@ services:
       - "8080:8080"
       - "4000:4000"
     environment:
-      - "EMITTER_LICENSE=PfA8INZBBcmxS4O3fj09pXaqZTJIwvJbJqFRpPmOsANZu75CEOIzfIuu3E3k-Q_wB2rCkv6qgOG3A57Q_OwPAQ:3"
-      # associated secret key Llo99jIV4vmN0BqJJO1Z5Y055DsPHnIS
+      - "EMITTER_LICENSE=PfA8IMohuWbuFm_jN3gegdx8c8Nqrd8CE9SYGS1MHOtUP8irEFdOS2nfL9SkZia6RUMBreHo_ZnwCPCgjoIMAQ:3"
+      # associated secret key bPcKwpuIP3gjjAKqb9EaAJYJnilISQhJ
```
[the remark this answers is not in the document]
Comment (gfanton) [reply tool/emitter-server/docker-compose.yml:10]: yes it's only for test purpose, it's not deploy in production

File: go/internal/tinder/service.go (hunk 3 above)
Comment (gfanton) [reply go/internal/tinder/service.go:141]: it should be used in a near future with prometheus metric

**Review by D4ryl00 — APPROVED** [review]
LGTM 👍

**Pull request merged**